internal/types/compare.go

```diff
@@ -259,6 +263,26 @@ func compareNumbers(a float64, b int64) CompareResult {
 	return CompareResult(bigA.Cmp(bigB))
 }
 
+// compareDocuments checks if two documents are equal.
+func compareDocuments(a, b *Document) CompareResult {
```

Comment:
we have a function that checks if two documents are equal to each other, it's called `matchDocuments`. But since I plan to work on the `compareDocuments` function, you can leave your made function, and delete the match.go file where the `matchDocuments` is located

Reply:
Well, there is still too much depending on matchDocuments, so it can't be deleted because of backward compatibility